### ISSUE TITLE
SC2174: should not warn about mkdir -p -m 0755 ../foo

### ISSUE THREAD

**Issue by eldering** [koalaman/shellcheck#854]
In a shell script I am using
```
#!/bin/sh
mkdir -p -m 0711 ../bin
```
to make sure that the directory `bin` exists with the right permissions. I use `-p` to make sure `mkdir` does not give an error when the directory already exists.

Shellcheck version 0.4.4 generates a warning here:
```
../judge/testcase_run.sh:172:10: warning: When used with -p, -m only applies to the deepest directory. [SC2174]
```
Although I realize this is a corner case, I think that any composite path where all but the last element consists of `.` or `..` should not trigger this warning.

**Comment by nihilus** [comment]
I beg to differ... makes totally perfect sense to warn about since people tend to do wrong. Myself would have done the wrong assumptions here... I started using the shell back on OSF/1 over 20 years ago, so I should have enough experience.

If you dont like an error you should programattically disable it in a comment.

**Comment by eldering** [comment]
Well, I'd say that intuitively one might think that the permissions apply to all newly created directories. Since `.` and `..` clearly already exist, it seems logical to me that a user would not assume that the permissions apply to these. However, if the general opinion is otherwise, please close this issue. (Indeed, I did already add an annotation to disable this particular check.)

**Comment by koalaman** [comment]
I agree that this shouldn't warn. Your wouldn't expect `-p` to create the directory `..` or `.`

**Comment by nihilus** [comment]
Yeah, however as I read the warning extending this to 'mkdir -p -m 0711 ../bin/laden' will not effect ../bin however if it doesnt exists it will be created but not with the mode specified.

**Comment by mrshu** [comment]
I've put together a PR that would fix this behavior for `..` and `.` (#904).

Note that `mkdir -p -m 0711 ../bin/laden` warns (if bin is created it does not have its mode set to 0711) while `mkdir -p -m 0711 ../bin` or `mkdir -p -m 0711 ./bin` does not, as per @koalaman's comment.
